### PR TITLE
Fix discarding chunk of data when write ends exactly in the end of the buffer.

### PR DESCRIPTION
**Pull request by KOLANICH**
Before: before
After: after

### CODE REVIEW

File: lfbb_cpp/inc/lfbb_impl.hpp [hunk 1]
```diff
@@ -90,9 +90,6 @@ void LfBb<T, size>::WriteRelease(const size_t written) {
 
     /* Increment the write index and wrap to 0 if needed */
```
Comment (DNedic) [inline lfbb_cpp/inc/lfbb_impl.hpp]: Can you fix the comment here?

File: lfbb_cpp/inc/lfbb_impl.hpp [hunk 2]
```diff
@@ -101,6 +98,10 @@ void LfBb<T, size>::WriteRelease(const size_t written) {
         i = w;
     }
 
+    if (w == size) {
```
Comment (DNedic) [inline lfbb_cpp/inc/lfbb_impl.hpp:102]: And add a comment here.

**Review by DNedic — CHANGES_REQUESTED** [review]
Thanks for the changes LGTM.

Just to summarize when this would happen in my understanding:
1. A write happens to write to the end of the buffer, `r` is higher than 0 but less than `w`
2. Because w is zeroed before i is set to w, now the data between i and size is considered invalid

Can you add a test for this?

File: lfbb_cpp/inc/lfbb_impl.hpp (hunk 1 above)
Comment (KOLANICH) [reply lfbb_cpp/inc/lfbb_impl.hpp]: Fixed.

File: lfbb_cpp/inc/lfbb_impl.hpp (hunk 2 above)
Comment (KOLANICH) [reply lfbb_cpp/inc/lfbb_impl.hpp:102]: Fixed.

**Pull request merged**